float32 target_velocity
float32 current_velocity
float32 demanded_velocity
